File: app/src/main/java/com/rokudo/xpense/data/retrofit/TokenInterceptor.java
package com.rokudo.xpense.data.retrofit;

import androidx.annotation.NonNull;

import com.rokudo.xpense.utils.GoCardlessUtils;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class TokenInterceptor implements Interceptor {

    @NonNull
    @Override
    public Response intercept(Chain chain) throws IOException {

        //rewrite the request to add bearer token
        Request newRequest = chain.request().newBuilder()
                .header("Authorization", "Bearer " + GoCardlessUtils.TOKEN_VAL)
                .build();

        return chain.proceed(newRequest);
    }
}

File: app/src/main/java/com/rokudo/xpense/data/retrofit/TokenInterceptor.kt
package com.rokudo.xpense.data.retrofit

import com.rokudo.xpense.utils.GoCardlessUtils
import okhttp3.Interceptor
import okhttp3.Response

class TokenInterceptor : Interceptor {
    override fun intercept(chain: Interceptor.Chain): Response {
        val newRequest = chain.request().newBuilder()
            .header("Authorization", "Bearer ${GoCardlessUtils.TOKEN_VAL}")
            .build()
        return chain.proceed(newRequest)
    }
}

